traversal/Traversal.java

```diff
@@ -136,136 +167,208 @@ public int hashCode() {
             return Objects.hash(iids, values);
         }
 
-        public static class Value {
+        public static class Value<T> implements Comparable<Value<?>> {
```

Comment:
we type the `Value`  and introduce subtypes for `String/Long/DateTime/Regex/Double/Boolean` to clean up the usages of this class in generified predicate execution